config/config.go

```diff
@@ -531,6 +533,20 @@ func adjustSourceWithMap(sourceMap map[string]string, source string, modulePath
 		}
 		moduleSubdir = moduleSubdirFromUrl
 	}
+	// if source path contains "?ref=", reconstruct module dir using "//"
+	if strings.Contains(sourcePath, "?ref=") && moduleSubdir != "" {
```

Comment:
wouldn't it be easier to split by `//`? convention over configuration approach.

Reply:
Wouldn't it be better to move this logic to `util.JoinTerraformModulePath` ?